examples/mem-correctness-test/src/main.rs

```diff
@@ -0,0 +1,279 @@
+    (get_free_heap_size_in_bytes() as f64 / heap_size() as f64) * 100.0
+}
+
+fn get_random_num(start: usize, end: usize) -> usize {
+    let mut rng = rand::thread_rng();
+    rng.gen_range(start..=end)
+}
+
+fn wait_per_thread(pair_clone: Arc<(Mutex<bool>, Condvar)>) {
+    let (lock, cvar) = &*pair_clone;
+    let mut started = lock.lock().unwrap();
+    while !*started {
+        started = cvar.wait(started).unwrap();
+    }
+    drop(started);
+}
+
+fn wakeup_all_child_threads(pair_clone: Arc<(Mutex<bool>, Condvar)>) {
+    let (lock, cvar) = &*pair_clone;
+    let mut started = lock.lock().unwrap();
+    *started = true;
+    cvar.notify_all();
+    drop(started);
+}
+
+fn traverse_and_check_buffer(buf: &Vec<usize>, magic_num_to_check: usize) {
+    let num_indices_checks = get_random_num(1, MAX_INDEX_CHECKS_PER_BUFFER);
+    for _i in 1..=num_indices_checks {
+        /* Check for random indices and number of such indices is  num_indices_checks
+         * Please note that depending on the number random number generator, we
+         * can check for the same index multiple times. We could have checked
+         * for all the indices but that would be too time consuming
+         */
+        if buf[get_random_num(0, buf.len() - 1)] != magic_num_to_check {
+            panic!("Corruption detected in traverse_and_check_buffer");
+        }
+    }
+}
+
+fn select_and_check_random_buffer_contents(
+    array_of_vectors: &Vec<Vec<usize>>,
+    magic_num_to_check: usize,
+) {
+    /* This function selects a random number of buffers and then calls
+     * traverse_and_check_buffer which checks random contents of the set of
+     * randomly selected buffers
+     */
+    let num_active_vectors = array_of_vectors.len();
+    if num_active_vectors > 0 {
+        let random_buffer_check_count = get_random_num(1, MAX_BUFFER_CHECKS_PER_THREAD_ITERATION);
+        for _i in 1..=random_buffer_check_count {
+            let random_buffer_index_to_check = get_random_num(0, array_of_vectors.len() - 1);
+            traverse_and_check_buffer(
+                &(array_of_vectors[random_buffer_index_to_check]),
+                magic_num_to_check,
+            );
+        }
+    }
+}
+
+fn delete_random_buffer(array_of_vectors: &mut Vec<Vec<usize>>) {
+    let num_active_vectors = array_of_vectors.len();
+    if num_active_vectors > 0 {
+        let random_index_to_delete = get_random_num(0, array_of_vectors.len() - 1);
+        let len = &array_of_vectors[random_index_to_delete].len() * mem::size_of::<i32>();
+        array_of_vectors.remove(random_index_to_delete);
+        update_occupied_heap_size_on_delete(len);
+    }
+}
+
+fn add_new_buffer(array_of_vectors: &mut Vec<Vec<usize>>, magic_num_to_check: usize) {
+    /* This function assumes that the percentage of free heap space is more
+     * than MIN_ALLOWED_FREE_HEAP_PERCENTAGE
+     */
+
+    let random_size = get_random_num(mem::size_of::<usize>() * 2, PER_THREAD_PER_BUFFER_MAX_SIZE)
```

Comment:
Can the minimum be 1 byte?

Reply:
I initially tried with 1 byte but since the vector is of of u64, I kept the minimum size to be 16 bytes